hypervisor/virtualbox/virtualbox.go

```diff
@@ -114,6 +129,29 @@ func (*virtualbox) Modify(vmName string, cpus int, mem int) error {
 	return nil
 }
 
+func (*virtualbox) GetVMProperty(vmName, property string) (*string, error) {
```

Comment:
so this method depends on 
https://docs.oracle.com/en/virtualization/virtualbox/6.0/user/guestadd-guestprops.html

LGTM. hopefully this works for bridge network

Reply:
yep, this works for bridge network, also i need to test this on a linux/unix machine so dont merge this yet, I have marked this as draft PR, will update once i test it on unix

Reply:
this does not work with unix/linux systems

apparantly the guestProperty get command only works for windows machines on powershell (i tried it in git bash and it failed there as well) for linux/unix systems we have to use guestProperty enumerate and then parse that object and grep ip address field from the string

updating this PR with new code

Reply:
So enumerate works for both Windows and Linux?

Reply:
yes enumerate works for both tested this on all platform mac windows and linux